### DISCUSSION TITLE
Problem with EPoll wait

### DISCUSSION THREAD

**Discussion by mariussafelog** [eclipse-milo/milo#1258] [General]
GC blocks my application because of native function in Epoll.
Anyone else had the problem? A fix could by having the possibility to use select(long timeout) with timeout.

Dump:
[dump.txt](https://github.com/eclipse/milo/files/14977489/dump.txt)

I found this:
"One problem that is actively being worked on is removing the need to lock out garbage collection when Java objects are used in native code: Java threads triggering a garbage collection must wait until no other regions are holding references to Java objects in native code. In the worst cases, native code may block garbage collection for minutes. This can lead to software developers choosing to not use native code at all, affecting throughput adversely. With the changes suggested in [JEP 423 (Region pinning for G1)](https://openjdk.java.net/jeps/423), this will become a nonissue for the G1 GC."

hier : https://blogs.oracle.com/javamagazine/post/java-garbage-collectors-evolution

**Comment by kevinherron** [comment]
@mariussafelog try a newer version of Java or perhaps report this to Azul or upstream with the entire crash dump file.
